EED_Affiliate_WP.module.php

```diff
@@ -240,4 +242,30 @@ protected static function _maybe_initiate_affiliate_tracking(EE_Transaction $tra
 
         $awp->visits->update($awp->tracking->get_visit_id(), array( 'referral_id' => $referral_id ), '', 'visit');
     }
+
+
+
+    /**
+     * Gets the base amount to calculate the referral amount from. Usually the Transaction Total.
+     *
+     * @param EE_Transaction $transaction
+     * @param Affiliate_WP   $awp
+     * @return float
+     */
+    public static function getInvoiceAmount(
+        EE_Transaction $transaction,
+        Affiliate_WP $awp
+    ) {
+        if ($awp->settings->get('exclude_tax')) {
+            $amount = $transaction->total() - $transaction->tax_total();
```

Comment:
I think this will work, but I'd be inclined to instead directly grab what we previously calculated the pretax total to be. Doing a subtraction here with two floats can potentially return a different amount.

```suggestion
            $amount = EEH_Line_Item::get_pre_tax_subtotal($transaction->total_line_item())->total();
```